pmd-scala-modules/pmd-scala-common/src/main/java/net/sourceforge/pmd/cpd/ScalaTokenizer.java

```diff
@@ -125,7 +137,13 @@ Token getNextToken() {
                 token = tokenIter.next();
             } while (token != null && skipToken(token) && tokenIter.hasNext());
 
-            return token;
+            result = new ScalaTokenAdapter(token, previousComment);
+
+            if (isComment(token)) {
+                previousComment = result;
+            }
+
+            return result;
```

Comment:
If the token is a comment it should actually not be returned, to implement this interface correctly. This is why your CPD-OFF comments are not filtered out (arguably all comments should actually be filtered out).

Reply:
rewritten to skip comments